### PR TITLE
Grades in studies matrix

### LINKED ISSUE #2944: As a student I want to see the grades of each completed course in "studies" view
#### Details

 - Show the abbreviation of the grade inside each node in the matrix

### PR DESCRIPTION
**Pull request by ilmoeuro**
** Fix #2944

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java [hunk 1]
```diff
@@ -224,14 +231,30 @@ public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString)
...
+
+                SchoolDataIdentifier gradingScaleIdentifier = workspaceAssessment.getGradingScaleIdentifier();
+                if (gradingScaleIdentifier == null) {
+                  break;
+                }
+                SchoolDataIdentifier gradeIdentifier = workspaceAssessment.getGradeIdentifier();
+                if (gradeIdentifier == null) {
+                  break;
+                }
+                GradingScale gradingScale = gradingController.findGradingScale(gradingScaleIdentifier);
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: These two will most likely cause a performance issue by queries to Pyramus. Anything to optimize this with?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java [hunk 2]
```diff
@@ -224,14 +231,30 @@ public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString)
...
+                GradingScaleItem gradingScaleItem = gradingController.findGradingScaleItem(gradingScale, gradeIdentifier);
+                if (!"".equals(gradingScaleItem.getName())) {
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: I'd prefer StringUtils.isEmpty.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java [hunk 3]
```diff
@@ -224,14 +231,30 @@ public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString)
...
+                  grade = gradingScaleItem.getName().substring(0, 1);
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: Doesn't seem optimal but I guess has to do as the interface has no better alternative for the "grade code".

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java (hunk 2 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: StringUtils.isBlank?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java (hunk 3 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: If you get a 10, it becomes 1?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java (hunk 1 above)
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java]: I'll implement a cache

**Comment by ilmoeuro** [comment]
** Fixed a serious bug, please re-review

**Pull request merged**